src/shared/components/oncoprint/makeGeneticTrackTooltip.tsx

```diff
@@ -0,0 +1,693 @@
+            profiledGenePanelCounts,
+            notProfiledGenePanelCounts,
+            profiledMolecularProfileCounts,
+            notProfiledMolecularProfileCounts,
+            allProfiledCount,
+            noneProfiledCount,
+        } = getProfileCounts(
+            dataUnderMouse,
+            molecularProfileMap,
+            alterationTypesInQuery
+        );
+
+        const profiledGenePanelEntries = profiledGenePanelCounts.entries();
+        const notProfiledGenePanelEntries = notProfiledGenePanelCounts.entries();
+
+        if (
+            profiledGenePanelEntries.length ||
+            notProfiledGenePanelEntries.length
+        ) {
+            ret.append('Gene Panels: ');
+            let needsCommaFirst = false;
+            for (const entry of profiledGenePanelEntries) {
+                if (needsCommaFirst) {
+                    ret.append(',&nbsp;');
+                }
+                needsCommaFirst = true;
+                ret.append(
+                    makeGenePanelPopupLink(
+                        entry.key[0],
+                        true,
+                        dataUnderMouse.length > 1 ? entry.value : undefined
+                    )
+                );
+            }
+            for (const entry of notProfiledGenePanelEntries) {
+                if (profiledGenePanelCounts.has(...entry.key)) {
+                    // only add again, as "not profiled", if we didn't already add it as "profiled"
+                    continue;
+                }
+                if (needsCommaFirst) {
+                    ret.append(',&nbsp;');
+                }
+                needsCommaFirst = true;
+                ret.append(
+                    makeGenePanelPopupLink(
+                        entry.key[0],
+                        false,
+                        dataUnderMouse.length > 1 ? entry.value : undefined
+                    )
+                );
+            }
+            ret.append('<br>');
+        }
+
+        // Molecular profile coverage
+        const profiledInEntries = profiledMolecularProfileCounts.entries();
+        const notProfiledInEntries = notProfiledMolecularProfileCounts.entries();
+
+        // only show specifics if not all profiled or all unprofiled
+        if (allProfiledCount === dataUnderMouse.length) {
+            ret.append(
+                'Profiled in all selected molecular profiles.' +
+                    (dataUnderMouse.length > 1 ? ` (${allProfiledCount})` : '')
+            );
+            ret.append('<br>');
+        } else if (noneProfiledCount === dataUnderMouse.length) {
+            // the moused over sample/case is not profiled
+            if (
+                profiledGenePanelEntries.length ||
+                notProfiledGenePanelEntries.length
+            ) {
+                // there is assumption that if the above condition is true
+                // the gene must be a fusion partner of a gene which IS profiled
+                ret.append(
```

Comment:
@inodb @jjgao i started this PR to try to bring clarity to this oncoprint tooltip code.  in particular i was trying to understand this code here.  @inodb you noticed this the other day that the oncoprint tooltip reported a sample/gene combination is a "probable fusion partner".   there is an assumption in the code here that i don't fully understand.   maybe you could help.  @jjgao i vaguely remember some long ago discussion with @kalletlak about this.

Reply:
![image](https://user-images.githubusercontent.com/186521/182715342-df316c58-3387-45aa-aeb3-81e05f370fa9.png)